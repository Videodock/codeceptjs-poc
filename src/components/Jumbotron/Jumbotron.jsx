import React from 'react';
import * as PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import { makeStyles } from '@material-ui/styles';
import { Typography } from '@material-ui/core';
import images from '../../theme/images';
import styles from './Jumbotron.styles';

const useStyles = makeStyles(styles);

export default function Jumbotron ({
  children,
  image,
  text: {
    prefixTitle,
    primaryTitle,
    secondaryTitle,
    introductionParagraph,
  },
  campaign,
  author,
  digitalDocument,
  isCampaign,
}) {
  const classes = useStyles();

  const renderContent = device => (
    <div className={classNames(classes[device], { [classes.campaign]: isCampaign })}>
      {!isCampaign && prefixTitle && (
        <div className={classes.prefixTitleHome}>
          <img className={classes.logoIcon} src={images.logoIcon} alt="" />
          <Typography variant="subtitle2">
            {prefixTitle}
          </Typography>
        </div>
      )}
      {isCampaign && (
        <div className={classes.prefixTitleCampaign}>
          <span className={classes.avatar} >
            <img src={images.avatarOne} alt="" />
          </span>
          <Typography variant="caption">
            <Link to="#">{author}</Link>
            {prefixTitle}
          </Typography>
        </div>
      )}
      {primaryTitle && (
        <Typography variant="h1">
          {primaryTitle}
        </Typography>
      )}
      <Typography variant="h2">
        {secondaryTitle ? secondaryTitle : <span className={classes.compositionTitle}>{campaign?.name}</span>}
      </Typography>
      <Typography gutterBottom>
        {isCampaign ? campaign?.description : introductionParagraph}
      </Typography>
      {children}
    </div>
  );

  return (
    <header>
      <div className={classNames(classes.root, { [classes.campaign]: isCampaign })}>
        {renderContent('desktop')}
        {!isCampaign && (<img className={classes.image} src={image} alt="" />)}
        {isCampaign && (
          <div className={classes.score}>
            <Typography variant="h3">
              {digitalDocument?.title}
            </Typography>
            <img className={classes.scoreImage} src={image} alt="" />
          </div>
        )}
      </div>
      {renderContent('mobile')}
    </header>
  );
}

Jumbotron.propTypes = {
  image: PropTypes.string,
  text : PropTypes.exact({
    prefixTitle          : PropTypes.string,
    primaryTitle         : PropTypes.string,
    secondaryTitle       : PropTypes.string,
    introductionParagraph: PropTypes.string,
  }),
  campaign       : PropTypes.object,
  digitalDocument: PropTypes.object,
  author         : PropTypes.string,
  isCampaign     : PropTypes.bool,
};
