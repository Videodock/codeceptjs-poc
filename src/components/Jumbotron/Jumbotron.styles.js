import { createStyles } from '@material-ui/styles';

export default ({ spacing, typography }) => createStyles({
  root: {
    position  : 'relative',
    width     : 1440,
    height    : 600,
    display   : 'flex',
    alignItems: 'center',
    background: 'linear-gradient(105deg, rgba(255,255,255,1) 65.9%, rgba(255,140,2,1) 66%, rgba(244,87,49,1) 100%)',
  },
  contentSection: {
    marginLeft  : 92,
    marginBottom: 170,
    width       : 475,
    height      : 50,
    zIndex      : 10,
  },
  subtitle: {
    display     : 'flex',
    alignItems  : 'center',
    marginBottom: spacing(0.5),
  },
  logoIcon: {
    width      : 15,
    height     : 14,
    marginRight: spacing(0.5),
  },
  buttonHero: {
    marginTop    : spacing(),
    padding      : '8px 32px',
    fontFamily   : typography.fontFamilyOpenSans,
    fontSize     : typography.pxToRem(16),
    letterSpacing: '0.66px',
  },
  image: {
    position: 'absolute',
    width   : '50%',
    height  : 'auto',
    right   : 92,
  },
});
