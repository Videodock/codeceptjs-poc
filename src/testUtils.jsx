import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
