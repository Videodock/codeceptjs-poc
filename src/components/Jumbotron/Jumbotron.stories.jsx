import React from 'react';
import { storiesOf } from '@storybook/react';
import Jumbotron from './Jumbotron';

storiesOf('Jumbotron', module)
  .add('basic', () => <Jumbotron />);
