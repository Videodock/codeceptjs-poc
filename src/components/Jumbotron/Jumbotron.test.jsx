import React from 'react';
import { render } from '../../testUtils';
import Jumbotron from './Jumbotron';

describe('<Jumbotron />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render((
      <Jumbotron
        text={{
          prefixTitle          : 'Subtitle',
          primaryTitle         : 'H1',
          secondaryTitle       : 'H2',
          introductionParagraph: 'Paragraph',
        }}
      />
    ));

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders all text props', () => {
    const { getAllByText } = render((
      <Jumbotron
        text={{
          slogan     : 'Slogan',
          description: 'Description',
        }}
      />
    ));

    expect(getAllByText('Slogan')).toBeTruthy();
    expect(getAllByText('Description')).toBeTruthy();
  });

  test('renders all text and campaign props if campaign', () => {
    const { getAllByText } = render((
      <Jumbotron
        campaign={{
          name: 'Mozart',
        }}
        digitalDocument={{
          title: 'Mahler',
        }}
        author="Annabel"
        campaignUrl="https://www.trompa.com"
        isCampaignPageHeader
      />
    ));

    expect(getAllByText('Annabel')).toBeTruthy();
    expect(getAllByText('Mahler')).toBeTruthy();
  });
});
