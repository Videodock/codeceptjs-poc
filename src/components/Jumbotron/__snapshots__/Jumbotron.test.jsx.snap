// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> renders and matches snapshot 1`] = `
<header>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-desktop-2"
    >
      <div
        class="makeStyles-subtitle-4"
      >
        <img
          alt=""
          class="makeStyles-logoIcon-5"
          src="[object Object]"
        />
        <h6
          class="MuiTypography-root MuiTypography-subtitle2"
        >
          Subtitle
        </h6>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h1"
      >
        H1
      </h1>
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        H2
        <span
          class="makeStyles-compositionTitle-6"
        >
          Requiem in D minor
        </span>
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Paragraph
      </p>
    </div>
    <img
      alt=""
      class="makeStyles-image-8"
    />
  </div>
  <div
    class="makeStyles-mobile-3"
  >
    <div
      class="makeStyles-subtitle-4"
    >
      <img
        alt=""
        class="makeStyles-logoIcon-5"
        src="[object Object]"
      />
      <h6
        class="MuiTypography-root MuiTypography-subtitle2"
      >
        Subtitle
      </h6>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h1"
    >
      H1
    </h1>
    <h2
      class="MuiTypography-root MuiTypography-h2"
    >
      H2
      <span
        class="makeStyles-compositionTitle-6"
      >
        Requiem in D minor
      </span>
    </h2>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    >
      Paragraph
    </p>
  </div>
</header>
`;
